src/lib.ts

```diff
@@ -683,3 +687,80 @@ export function printEventLog(): void {
     );
   }
 }
+
+const rateCache = new Map<string, Map<Monster, number>>();
+
+export function barfEncounterRate(options: {
+  snapper?: boolean;
+  snapperPhylum?: Phylum;
+  olfact?: Monster;
+  longCon?: Monster;
+  motif?: Monster;
+  turtle?: Monster;
+  monkeyPoint?: Monster;
+  humanity?: boolean;
+}): Map<Monster, number> {
+  const olfact = options.olfact ?? get("olfactedMonster");
+  const longCon = options.longCon ?? get("longConMonster");
+  const motif = options.motif ?? get("motifMonster");
+  const turtle = options.turtle ?? get("_gallapagosMonster");
+  const monkeyPoint = options.monkeyPoint ?? get("monkeyPointMonster");
+  const snapper = options.snapper ?? myFamiliar() === $familiar`Red-Nosed Snapper`;
+  const snapperPhylum = options.snapperPhylum ?? Snapper.getTrackedPhylum();
+  const humanity = options.humanity ?? have($effect`Ew, The Humanity`);
```

Comment:
imo i would refactor this as something like
```suggestion
  const DEFAULT_OPTIONS = {
    olfact: get("olfactedMonster"),
    longCon: get("longConMonster"),
    etc,
   };
   const { olfact, longCon, motif, etc } = { ...DEFAULT_OPTIONS, ...options };
```
but if that looks a lot don't do that

Reply:
a lot worse*